### PR TITLE
fix(pipedrive): incremental data is None

### PR DESCRIPTION
**Pull request by IlyaFaer**
Fixing a bunch of Pipedrive things:

1. let's use `"1970-01-01 00:00:00"` as a default for `since_timestamp`. that should fix the problem
2. let's drop the `Optional` from all Incremental declarations. they are not optional
3. we unified how we deal with datetimes/timestamps in our sources. please look ie. at `def zendesk_talk` and how `TAnyDateTime` is used together with `ensure_pendulum_datetime` (yeah pipedrive is using some non ISO date format... so part of the conversion function should stay)
4. Steini left this comment
```
# TODO: This check can be replaced with `update_time.start_out_of_range` in dlt 0.3.5
```
let's implement it

See the original [issue](https://github.com/dlt-hub/verified-sources/pull/298#issuecomment-1854468600)

### CODE REVIEW

File: sources/pipedrive/__init__.py [hunk 1]
```diff
@@ -184,17 +187,13 @@ def leads(
     )
     # Load leads pages sorted from newest to oldest and stop loading when
     # last incremental value is reached
-    last_value = update_time.last_value
```
Comment (IlyaFaer) [inline sources/pipedrive/__init__.py:187]: This value is no longer needed, `start_out_of_range` should do the job.

File: sources/pipedrive/helpers/__init__.py [hunk 2]
```diff
@@ -5,18 +5,6 @@
...
-def parse_timestamp(timestamp: Any) -> str:
-    if not timestamp:
-        return None
-    if isinstance(timestamp, str):
-        timestamp = pendulum.parse(timestamp)
-    assert isinstance(
-        timestamp, pendulum.DateTime
-    ), "since_timestamp must be a valid ISO datetime string or pendulum.DateTime object"
-    timestamp = timestamp.in_timezone("UTC")
-    return cast(str, timestamp.to_iso8601_string().replace("T", " ").replace("Z", ""))
```
Comment (IlyaFaer) [inline sources/pipedrive/helpers/__init__.py:17]: It's used only for Pipedrive, so it's no longer needed

Comment (rudolfix) [reply sources/pipedrive/helpers/__init__.py:17]: thanks for cleaning that up!

**Review by rudolfix — APPROVED** [review]
LGTM! thanks for cleaning that up!

**Review by rudolfix — CHANGES_REQUESTED** [review]
just one more thing: please resubmit this PR directly so all tests can run.

**Comment by rudolfix** [comment]
continued in #303

**Pull request closed without merging**